clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2077,6 +2077,25 @@ void Sema::finalizeSYCLDelayedAnalysis(const FunctionDecl *Caller,
   }
 }
 
+bool Sema::isArrayZeroInitialized(VarDecl *VD, bool CheckValueDependent) {
+  assert(getLangOpts().SYCLIsDevice &&
+         "Should only be called during device compilation");
+  if (VD->isInvalidDecl() || !VD->hasInit() || !VD->hasGlobalStorage())
+    return false;
+  bool zeroInitArray = false;
+  unsigned NumOfArrayInitElts;
+  APValue *Value = nullptr;
+  if (VD->getInit() && !VD->getInit()->isValueDependent())
+    Value = VD->evaluateValue();
+  if (Value && Value->isArray() && Value->hasArrayFiller()) {
+    NumOfArrayInitElts = Value->getArrayInitializedElts();
```

Comment:
Suggest changing this to:

```
if (Value->getArrayInitializedElts() == 0)
  return true;
...

return false;
```

and remove both zeroInitArray and NumOfArrayInitElts variables.

Reply:
Done.